src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java

```diff
@@ -33,4 +39,35 @@ public FlashcardSaveResponseDto saveFlashcard(FlashcardDto flashcardDto) {
             throw new EntityNotCreatedException("something went wrong");
         }
     }
+
+    public FlashcardSaveResponseDto updateFlashcard(FlashcardDto flashcardDto, Long id) {
+        Flashcard flashcardFound = flashcardService.findById(id);
+
+        CompareQuestionTagsSets tagsToUpdate = compareQuestionTagSets(flashcardFound.getQuestionTagsSet(), flashcardDto.getTagsList());
+        questionTagService.updateFlashcardSet(tagsToUpdate, flashcardFound);
+
+        Set<QuestionTag> questionTagsToAddSet = questionTagService.getQuestionTagsSet(tagsToUpdate.getTagsNamesToAdd());
+        Set<QuestionTag> questionTagsToRemoveSet = questionTagService.getQuestionTagsSet(tagsToUpdate.getTagsNamesToRemove());
+        flashcardFound.getQuestionTagsSet().addAll(questionTagsToAddSet);
+        flashcardFound.getQuestionTagsSet().removeAll(questionTagsToRemoveSet);
+
+        flashcardFound.setChanges(flashcardDto);
+        flashcardService.saveFlashcard(flashcardFound);
+
+        return modelMapper.map(flashcardFound, FlashcardSaveResponseDto.class);
+    }
+    private CompareQuestionTagsSets compareQuestionTagSets(Set<QuestionTag> questionTagsSetBeforeChanges, Set<String> questionTagsNamesSetActual) {
```

Comment:
What would you say about converting this function into `CompareQuestionTagsSets`'s constructor?

Reply:
I like it :1st_place_medal: